ethereum/dex/trades/insert_xsigma.sql

```diff
@@ -0,0 +1,169 @@
+CREATE OR REPLACE FUNCTION dex.insert_xsigma(start_ts timestamptz, end_ts timestamptz=now(), start_block numeric=0, end_block numeric=9e18) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+WITH rows AS (
+    INSERT INTO dex.trades (
+        block_time,
+        token_a_symbol,
+        token_b_symbol,
+        token_a_amount,
+        token_b_amount,
+        project,
+        version,
+        category,
+        trader_a,
+        trader_b,
+        token_a_amount_raw,
+        token_b_amount_raw,
+        usd_amount,
+        token_a_address,
+        token_b_address,
+        exchange_contract_address,
+        tx_hash,
+        tx_from,
+        tx_to,
+        trace_address,
+        evt_index,
+        trade_id
+    )
+
+    SELECT
+        dexs.block_time,
+        erc20a.symbol AS token_a_symbol,
+        erc20b.symbol AS token_b_symbol,
+        token_a_amount_raw / 10 ^ erc20a.decimals AS token_a_amount,
+        token_b_amount_raw / 10 ^ erc20b.decimals AS token_b_amount,
+        'xSigma' AS project,
+        '1' AS version,
+        'DEX' AS category,
+        COALESCE(trader_a, tx."from") AS trader_a,
+        trader_b,
+        token_a_amount_raw,
+        token_b_amount_raw,
+        GREATEST(
+            token_a_amount_raw / 10 ^ pa.decimals * pa.price,
+            token_b_amount_raw / 10 ^ pb.decimals * pb.price
+        ) AS usd_amount,
+        token_a_address,
+        token_b_address,
+        exchange_contract_address,
+        tx_hash,
+        tx."from" AS tx_from,
+        tx."to" as tx_to,
+        trace_address,
+        evt_index,
+        row_number() OVER (PARTITION BY tx_hash, evt_index, trace_address) AS trade_id
+
+    FROM (
+      
+      -- SigThreePool_evt_TokenExchange
+
+        SELECT
+          swaps."evt_block_number" as block_time,
```

Comment:
Thank you @Zerquix18. Great job. Definitely a nice challenge for your first Dune contribution.

In general, token_a is the token bought. Can you please make the following changes?
```
-          swaps."evt_block_number" as block_time,
-          NULL::bytea as trader_a,
-          swaps."buyer" as trader_b,
+          swaps."evt_block_time" as block_time,
+          swaps."buyer" as trader_a,
+          NULL::bytea as trader_b,
```

Reply:
Hi @masquot 
Thanks for your feedback

I pushed a new commit with the changes. Let me know what you think

Reply:
Looks good to me, thank you.